### PR TITLE
chore: use self-hosted runner for sim tests

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

sim tests are consistently flaky

**Description**

use newly provisioned self-hosted runner for sim tests

### CODE REVIEW

File: .github/workflows/test-sim-merge.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ env:
 jobs:
   sim-merge-tests:
     name: Sim merge tests
-    runs-on: buildjet-4vcpu-ubuntu-2204
+    runs-on: Sim-Test-Github-runner
```
Comment (matthewkeil) [inline .github/workflows/test-sim-merge.yml:27]: I see this in the runners tab. How does it wire through to point to Hetzner?

**Review by matthewkeil — APPROVED** [review]
Cool!! Should we put a note somewhere about where the self hosted runner is located so the user can go find the logs and what not?  Not sure where the best place to put that is though...

**Comment by dapplion** [comment]
Note that external contributions won't run sim tests anymore, would be good to note that somewhere or make the host a dynamic

**Comment by nflaig** [comment]
Before going ahead with this, should there be another attempt to make sim tests stable without changing runners? Based on my observation before https://github.com/ChainSafe/lodestar/pull/6163 they at least passed if the CI was not overloaded but now they fail basically all the time. If it is really just the unknown block sync node we might be able to fix it.

Where I think we definitely need a self-hosted runner is for the benchmarks, they take around 45 minutes at the moment and produce random false positives all the time.

**Comment by wemeetagain** [comment]
- Closing this PR for now since https://github.com/ChainSafe/lodestar/pull/6312 re-stabilized the sim tests
- We can open a similar PR for the benchmarks

**Pull request closed without merging**